File: hero-parts-platform/src/main/java/com/hero/parts/model/Part.java
package com.hero.parts.model;

import jakarta.persistence.*;
import lombok.*;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.List;

@Entity
@Table(name = "parts")
@Getter @Setter @NoArgsConstructor @AllArgsConstructor @Builder
public class Part {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, unique = true, length = 50)
    private String sku;

    @Column(nullable = false, length = 500)
    private String name;

    @Column(name = "hindi_name", length = 500)
    private String hindiName;

    @Column(columnDefinition = "TEXT")
    private String description;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "category_id")
    private Category category;

    @Column(precision = 10, scale = 2)
    private BigDecimal price;

    @Column(precision = 10, scale = 2)
    private BigDecimal mrp;

    @Column(length = 20)
    private String unit;

    @Column(name = "compatible_models", length = 500)
    private String compatibleModels;

    @Column(name = "eshop_url", length = 500)
    private String eshopUrl;

    @Column(name = "image_url", length = 500)
    private String imageUrl;

    @Column(name = "in_stock")
    private Boolean inStock;

    @Column(name = "stock_qty")
    private Integer stockQty;

    @OneToMany(mappedBy = "part", cascade = CascadeType.ALL, orphanRemoval = true, fetch = FetchType.LAZY)
    private List<SearchAlias> aliases;

    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;

    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }
}

File: hero-parts-frontend/src/types/part.ts
export type AliasType = 'HINDI' | 'SLANG' | 'PHONETIC' | 'SYMPTOM' | 'COLOR' | 'MODEL_SLANG'

export interface PartAlias {
  id: number
  alias: string
  aliasType: AliasType
  language: string
}

export interface Part {
  id: number
  sku: string
  name: string
  hindiName: string | null
  description: string | null
  categoryId: number
  categoryName: string
  price: number
  mrp: number
  unit: string
  compatibleModels: string
  eshopUrl: string | null
  imageUrl: string | null
  inStock: boolean
  stockQty: number
  aliases: PartAlias[]
}

export interface Category {
  id: number
  name: string
  description: string | null
  parentId: number | null
  parentName: string | null
  subCategories: Category[]
  partCount: number
}
